Add an example of FastViT model (Infernece)

cfgfung opened the pull request:
# What does this PR do?
Added the example code of the validated FastViT.

Original performance:
n_iterations: 20
Total latency (ms): 451.754093170166
Average latency (ms): 22.5877046585083

BF16 + Graph mode:
n_iterations: 20
Total latency (ms): 47.97697067260742
Average latency (ms): 2.398848533630371

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

yeonsily commented:
@cfgfung Is this the same change with https://github.com/huggingface/optimum-habana/pull/814?

yeonsily commented on examples/object-segementation/ClipSeg/run_example.py:
```diff
@@ -0,0 +1,114 @@
+        model = ht.hpu.wrap_in_hpu_graph(model)
+
+    autocast = torch.autocast(device_type="hpu", dtype=torch.bfloat16, enabled=args.bf16)
+    model.to("hpu")
```
don't we need model.eval()?

cfgfung replied:
torch.no_grad() is used in line 87. The model will behave the same as model.eval().

cfgfung commented:
> @cfgfung Is this the same change with #814?

@yeonsily This is another example. Look similar but not the same.

cfgfung commented:
Add GPU Benchmark result:

A100 - 80GB , CUDA 12.2 , NVIDIA-SMI 535.54.03
With BF16 Autocast:
n_iterations: 20
Total latency (ms): 138.54217529296875
Average latency (ms): 6.9271087646484375

cfgfung commented:
Hi,

Some updates here:
- Tested with 'make style' to align the code styles.
- Added and updated related readmes.
- Intentionally not adding test cases as this is a simple example
- This is a TIMM model and does not belong to Transformer module. The current existing codes in run_image_classification.py are not compatible with TIMM model. Therefore I separate this into another file. run_timm_example.py.

libinta commented:
@cfgfung  can you rebase and provide the latest test result. thanks

cfgfung commented:
> @cfgfung can you rebase and provide the latest test result. thanks

Hi, 
I have applied rebase on this branch and added test cases.

![ViT_TestCases](https://github.com/huggingface/optimum-habana/assets/15855337/ab749b86-607d-45e6-8215-d2cafde774af)

yeonsily commented:
LGTM. Just can you please make sure if the CI pass on the synapse 1.16 release?

cfgfung commented:
> LGTM. Just can you please make sure if the CI pass on the synapse 1.16 release?

Hi,
I have ran the test on Synapse 1.16 and it passed.

![image](https://github.com/huggingface/optimum-habana/assets/15855337/c4dd5c05-7661-4541-a47e-db2e771c9c3e)

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/optimum-habana/pr_826). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.
